### PR TITLE
example of having an application layer between the framework http han…

### PR DESCRIPTION
**Pull request by maleck13**
…dling and domain logic

@philbrookes  Here is basically what I was talking about with the hexagonal architecture stuff, main thing to notice here is how the http handler just decodes and then hands off control. Meaning any other form of transport could be easily implemented and handled in the same way. A message bus for example

### CODE REVIEW

File: deployHandler.go [hunk 1]
```diff
@@ -66,12 +98,12 @@ func (dc *DeployCloudAppPayload) validate() ([]string, error) {
...
 // NewDeployHandler creates a cloudApp controller.
-func NewDeployHandler(logger Logger, paasService PaaSService, namer Namespacer) DeployHandler {
+func NewDeployHandler(logger Logger, deployController controller.Deploy, namer Namespacer) DeployHandler {
 	return DeployHandler{
-		logger:      logger,
-		paasService: paasService,
-		namer:       namer,
-		objectNamer: rhmap.Service{},
+		logger:           logger,
+		deployController: deployController,
```
Comment (philbrookes) [inline deployHandler.go]: oh, this is a deployHandler with a handle to a deployController.

File: main.go [hunk 2]
```diff
@@ -34,12 +35,16 @@ func buildPaasService() openshift.Service {
...
+	return controller.NewDeployController(buildPaasService())
+}
+
 func buildSysHandler() SysHandler {
 	return SysHandler{}
 }
 
 func buildDeployHandler() DeployHandler {
-	return NewDeployHandler(logrus.StandardLogger(), buildPaasService(), &config.Conf{})
+	return NewDeployHandler(logrus.StandardLogger(), buildDeployController(), &config.Conf{})
```
Comment (philbrookes) [inline main.go]: as above, I had misread.

File: controller/deploy.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	CloudAppName() string
+	BuildConfigName() string
+	ServiceName() string
+	CloudAppGUID() string
+	Project() string
+	DomainName() string
+	UserName() string
+	Authentication() string
+	SourceLoc() string
+	SourceBranch() string
```
Comment (philbrookes) [inline controller/deploy.go]: This command needs environment vars too.

File: deployHandler.go (hunk 1 above)
Comment (maleck13) [inline deployHandler.go]: @philbrookes It doesn't the word controller doesn't mean a http handler. The deployHandler has a deployController the deployController's Job is to control and coordinate a deploy. It knows nothing about http

File: controller/deploy.go (hunk 3 above)
Comment (maleck13) [inline controller/deploy.go]: Yeah it is more just to give a concrete example of what I meant

File: deployHandler.go (hunk 1 above)
Comment (philbrookes) [inline deployHandler.go]: yeah, I initially thought this was a nested deployController, I hadn't noticed the different names. If DeployHandler is the part that knows about HTTP, should that be reflected in it's name?

**Pull request merged**